src/temporal/t.rast.aggregate.seasons/t.rast.aggregate.seasons.py

```diff
@@ -0,0 +1,219 @@
+    else:
+        out_strds = basename
+
+    seasons_name = ["spring", "summer", "autumn", "winter"]
+
+    # create new space time vector datasets one for each year to be used as sampler
+    for year in years:
+        season_vect = []
+        for seas in seasons_name:
+            name = f"sample_{seas}_{year}"
+            vect = VectorTopo(name)
+            vect.open("w")
+            point = Point(0, 0)
+            vect.write(point, cat=1)
+            vect.close()
+            map_layer = tgis.space_time_datasets.VectorDataset(f"{name}@{mapset}")
+            if seas == "spring":
+                extent = tgis.AbsoluteTemporalExtent(
+                    start_time=datetime(int(year), 3, 20),
+                    end_time=datetime(int(year), 6, 21),
+                )
+            elif seas == "summer":
+                extent = tgis.AbsoluteTemporalExtent(
+                    start_time=datetime(int(year), 6, 21),
+                    end_time=datetime(int(year), 9, 20),
+                )
+            elif seas == "autumn":
+                extent = tgis.AbsoluteTemporalExtent(
+                    start_time=datetime(int(year), 9, 20),
+                    end_time=datetime(int(year), 12, 21),
+                )
+            elif seas == "winter":
+                extent = tgis.AbsoluteTemporalExtent(
+                    start_time=datetime(int(year), 12, 21),
+                    end_time=datetime(int(year) + 1, 3, 20),
+                )
+            map_layer.set_temporal_extent(extent=extent)
+            season_vect.append(map_layer)
+        outsp = tgis.open_new_stds(
+            f"sample_seasons_{year}",
+            "stvds",
+            "absolute",
+            f"Season vector year {year}",
+            f"Season vector for the year {year}",
+            "mean",
+            dbif,
+            gscript.overwrite(),
+        )
+        tgis.register_map_object_list(
+            "vector",
+            season_vect,
+            outsp,
+            False,
+            None,
+            dbif,
+        )
+
+    process_queue = pymod.ParallelModuleQueue(int(nprocs))
+
+    # create t.rast.aggregate.ds module to be copied
+    mod = pymod.Module("t.rast.aggregate.ds")
+    mod.inputs.input = strds
+    mod.inputs.method = method
+    mod.inputs.basename = basename
+    mod.inputs.type = "stvds"
+    mod.flags.quiet = True
+    mod.flags.n = register_null
+
+    count = 0
+
+    # for each year calculate seasonal aggregation
+    for year in years:
+        mymod = copy.deepcopy(mod)
+        mymod.inputs.sample = f"sample_seasons_{year}@{mapset}"
+        mymod.outputs.output = f"{out_strds}_{year}"
+        process_queue.put(mymod)
+
+        if count % 10 == 0:
+            gscript.percent(count, len(years), 1)
+
+    # Wait for unfinished processes
+    process_queue.wait()
+
+    out_maps = []
+    # remove space time vector datasets
+    for year in years:
+        remod = pymod.Module("t.remove")
```

Comment:
Isn't the default for run_ True? See https://grass.osgeo.org/grass78/manuals/libpython/_modules/pygrass/modules/interface/module.html#Module

Reply:
Or is that only when other parameters are supplied? (Then disregard my comment about parallel processing.)